### DISCUSSION TITLE
Support for pyVista widget?

### DISCUSSION THREAD

**Discussion by gsal** [zauberzeug/nicegui#1366] [Ideas / Feature Requests]
I want to take advantage of VTK functionality.

So, I looked at trame, for sure, as the supposedly natural choice to get to VTK; but, to me, it seems a bit convoluted and too vtk-centric...so vtk-centric that for any other widgets typical of a dashboard application, one needs to visit Vuetify, look the widget docs in javascript, and 'translate' to python based on some rules in the trame doc itself...not cool. 

And, so, I rather learn nicegui as the all a round frontend for my applications and then bring in pyvista when needing vtk stuff. 

Is it possible?
Does anybody know how to properly include a [pyvista](https://docs.pyvista.org/) widget to load a vtk file, visualize and possible do some plane cutting and stuff?

**Comment by rodja** [comment]
For reference: in April there was a already an Q/A discussion about pyvista: https://github.com/zauberzeug/nicegui/discussions/745

**Reply by gsal** [reply]
Yes, I saw #745; but, it was closed and I did not see a way to re-open, so created this. 

Besides, #745 did not look like an answer when going the route of some exported html file; kind of hack-ish...I wouldn't even know the consequence of that as far as interactivity, callbacks etc. 

Thanks.

**Comment by falkoschindler** [comment]
Sure, that's probably possible. A pull request for a first proof-of-concept would be very welcome. 🙂

**Reply by gsal** [reply]
I hope you are not asking me...I am a user, not a developer ;-)

**Comment by liushuya7** [comment]
I really hope that nicegui can support pyvista soon!
